lasagne/tests/layers/test_conv.py

```diff
@@ -566,10 +593,17 @@ def test_with_nones(self, DummyInputLayer, input, kernel, output, kwargs):
                 filter_size=kernel.shape[2:],
                 W=kernel.transpose(1, 0, 2, 3),
                 **kwargs)
-        assert layer.output_shape == (None, output.shape[1], None, None)
+        if 'output_size' not in kwargs or isinstance(kwargs['output_size'],
+                                                     T.Variable):
+            assert layer.output_shape == (None, output.shape[1], None, None)
         actual = layer.get_output_for(input).eval()
         assert actual.shape == output.shape
         assert np.allclose(actual, output)
+        # Check get_output_shape_for for symbolic output
+        if 'output_size' in kwargs and isinstance(kwargs['output_size'],
+                                                  T.Variable):
+            assert all(el is None for el in
+                       layer.get_output_shape_for(input)[2:])
```

Comment:
You tested the symbolic case above already. What we'd need here would be a test for a nonsymbolic output size, to check if it's reported to the outside world.